pkgs/development/python-modules/mpris_server/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, emoji
+, pydbus
+, pygobject3
+, unidecode
+, setuptools
+,
+}:
+buildPythonPackage rec {
+  pname = "mpris_server";
+  version = "0.4.2";
+  format = "pyproject";
+
+  src = fetchPypi {
+    pname = "mpris_server";
+    inherit version;
+    hash = "sha256-p3nM80fOMtRmeKvOXuX40Fu9xH8gPgYyneXbUS678fE=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [
+    emoji
+    pydbus
```

Comment:
it looks like `pydbus` is not supported beyond python310. I just merged master in my branch and seems like default python is now 3.11 and build breaks with:
```
error: pydbus-0.6.0 not supported for interpreter python3.11
```

Reply:
Hmm. What should i do then? Hardcode python 310?

Reply:
> Hmm. What should i do then? Hardcode python 310?

somehow prevent a python > 3.10. not sure what the options are other than hardcoding but there is probably some way to specify an upper bound -- just not sure what it is, so hardcoding seems fine. i'll grep around a little.

Reply:
I can just use python310 package, don't worry.

Reply:
something like this in `.../package.nix` works for me.
```nix
{
...
, python310Packages
}:
let
  # pydbus from mpris_server is constrained to python310
  python3Packages = python310Packages;
in
python3Packages.buildPythonApplication {
...
```

Reply:
Why wouldn't i just use python310Packages instead of python3Packages everywhere

Reply:
> Why wouldn't i just use python310Packages instead of python3Packages everywhere

that works too. but at some point mpris_server or monophony might be updated to work with python 311 or beyond so the constraint is removed. the let/in also shows the constraint with a single comment. either way works.

Reply:
I see, thanks.

Reply:
I should be able to fix this tomorrow.

Reply:
Fixed